fuse/utils/ndict.py

```diff
@@ -162,84 +122,142 @@ def items(self) -> dict_items:
-        for sub_key in nested_key:
-            if isinstance(value, MutableMapping) and sub_key in value:
-                value = value.get(sub_key)
-            else:
-                raise NestedKeyError(key, self)
+    def is_prefix(self, key: str) -> bool:
+        """
+        return True iff the key is a *strictly* prefix of another key(s) in the NDict.
+        :param key:
+
+        Example:
+            >>> ndict = NDict()
+            >>> ndict["a.b.c"] = "d"
+            >>> ndict.is_prefix("a")
+            True
+            >>> ndict.is_prefix("a.b")
+            True
+            >>> ndict.is_prefix("a.b.c")
+            False    # STRICTLY PREFIX!
+        """
+        key_dot = key + "."
+        # iterate over all keys and looking for a match
+        for kk in self.keypaths():
+            if kk.startswith(key_dot):
+                return True
+
+        # a match wasn't found
+        return False
+
+    def get_sub_dict(self, key: str) -> NDict:
+        """
+        returns a copy of the "sub-dict" give a sub-key.
+
+        Example:
+            >>> ndict = NDict()
+            >>> ndict["a.b.c"] = "x"
+            >>> ndict["a.b.d"] = "y"
+            >>> ndict["a.b.e"] = "z"
+            >>> ndict.get_sub_dict("a.b")
+            {'c': 'x', 'd': 'y', 'e': 'z'}
+        """
+        res = NDict()
+        prefix_key = key + "."
+        suffix_key = None
+        for kk in self.keypaths():
+            if kk.startswith(prefix_key):
+                suffix_key = kk.replace(prefix_key, "", 1)
+                res[suffix_key] = self[kk]
+
+        if suffix_key is None and key not in self:
+            raise NestedKeyError(key, self)
 
-        return value
+        return res
 
     def __setitem__(self, key: str, value: Any) -> None:
         """
-        go over the the dictionary according to the path, create the nodes that does not exist
+        sets item in the dictionary.
+        if the item is a dict like object, it will parse the values using the delimiter "."
+
+        Example:
+            ndict = NDict()
+            ndict["a"] = {"b" : {"c" : 42}}
+
+            assert ndict["a.b.c"] == 42
+
         :param key: the keypath
         :param value: value to set
         """
-        # if value is dictionary add to self key by key to avoid from keys with delimeter "."
+        # if value is dictionary add to self key by key to avoid from keys with delimiter "."
         if isinstance(value, MutableMapping):
             for sub_key in value:
                 self[f"{key}.{sub_key}"] = value[sub_key]
             return
+        self._stored[key] = value
 
-        nested_key = key.split(".")
-        element = self._stored
-        for key in nested_key[:-1]:
-            if key not in element:
-                element[key] = {}
-            element = element[key]
+    def __delitem__(self, key: str) -> None:
+        """
+        :param key:
+        TODO should we delete both value and prefix ?
```

Comment:
@mosheraboh 

What do you think?

Currently we delete the value (and not the subdict!) 
if the value doesn't exists but the subdict does, we delete the subdict.

Reply:
ans (talked offline):
delete both